corehq/apps/reports/management/commands/prepare_projects_for_case_list_explorer.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime
+
+from django.core.management import BaseCommand
+
+from dimagi.utils.couch.database import iter_docs
+from corehq.apps.case_search.models import DomainsNotInCaseSearchIndex
+from corehq.apps.domain.models import Domain
+from corehq.apps.es import CaseES, CaseSearchES
+from corehq.pillows.case_search import CaseSearchReindexerFactory, domain_needs_search_index
+
+
+class Command(BaseCommand):
+    """Helper tool for preparing legacy projects on CommCare HQ instances for the General Release
+    of Case List Explorer. This is done by ensuring data is synced between the Elasticsearch CaseIndex
+    and CaseSearchIndex.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--project',
+            action='store',
+            dest='domain',
+            help='Specify a specific project to run data a integrity check and re-sync on.'
+        )
+
+    def handle(self, **options):
+        domain = options.pop('domain')
+        if domain is not None:
+            if DomainsNotInCaseSearchIndex.objects.count() > 0:
+                self.stdout.write("Please finish running the complete initial sync by "
+                                  "running this management command without the --project "
+                                  "argument.")
+                return
```

Comment:
This feels like an error? It would be good to conform to shell conventions and output errors to stderr while returning a non-zero return code.